metrics/api/src/main/java/io/helidon/metrics/api/Counter.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.metrics.api;
+
+/**
+ * Records a monotonically increasing value.
+ */
+public interface Counter extends Meter {
+
+    /**
+     * Updates the counter by one.
+     */
+    void increment();
+
+    /**
+     * Updates the counter by the specified amount which should be non-negative.
+     *
+     * @param amount amount to add to the counter.
+     */
+    void increment(double amount);
+
+    /**
+     * Returns the cumulative count since this counter was registered.
+     *
+     * @return cumulative count since this counter was registered
+     */
+    double count();
```

Comment:
why is double used for a counter? This should be integers by definition as I understand the concept of coutners

Reply:
If you look at Micrometer and OpenTelemetry metrics, `double` is what Micrometer uses for counters and OpenTelemetry exposes `LongCounter` and `DoubleCounter`. I do not think we want to expose two kinds of counters, so using `double` seemed to make sense to align with the likely implementations of our API.

Reply:
I have changed these to use `long` instead of `double` for counters, counts, etc.